internal/dao/dao_interfaces.go

```diff
@@ -2,10 +2,39 @@ package dao
 
 import (
 	"context"
+	"database/sql"
 	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/jmoiron/sqlx"
 )
 
-var GetAccountDao func(ctx context.Context) (AccountDao, error)
+// Transaction represents both database connection pool and a single transaction/connection
+type Transaction interface {
+	// ExecContext from database/sql library
+	ExecContext(ctx context.Context, query string, args ...interface{}) (sql.Result, error)
+	// PrepareContext from database/sql library
+	PrepareContext(ctx context.Context, query string) (*sql.Stmt, error)
+	// QueryContext from database/sql library
+	QueryContext(ctx context.Context, query string, args ...interface{}) (*sql.Rows, error)
+	// QueryRowContext from database/sql library
+	QueryRowContext(ctx context.Context, query string, args ...interface{}) *sql.Row
+
+	// QueryxContext from sqlx library
+	QueryxContext(ctx context.Context, query string, args ...interface{}) (*sqlx.Rows, error)
+	// QueryRowxContext from sqlx library
+	QueryRowxContext(ctx context.Context, query string, args ...interface{}) *sqlx.Row
+	// PreparexContext from sqlx library
+	PreparexContext(ctx context.Context, query string) (*sqlx.Stmt, error)
+	// GetContext from sqlx library
+	GetContext(ctx context.Context, dest interface{}, query string, args ...interface{}) error
+	// SelectContext from sqlx library
+	SelectContext(ctx context.Context, dest interface{}, query string, args ...interface{}) error
+}
+
+// A TxFn is a function that will be called with an initialized `Transaction` object
+// that can be used for executing statements and queries against a database.
+type TxFn func(Transaction) error
+
```

Comment:
This might look scary, but what this interface really does is it allows our DAO code to be completely independent of underlaying implementation. All DAO functions can simply use `Transaction` interface regardless if its coming either from a connection pool or from an explicit transaction. See below for how to use the TxFn function.